src/Resolve/graphtype.jl

```diff
@@ -241,21 +241,26 @@ mutable struct Graph
             uuid_to_name::Dict{UUID,String},
             reqs::Requires,
             fixed::Dict{UUID,Fixed},
-            verbose::Bool = false
+            verbose::Bool = false,
+            julia_version::Union{VersionNumber,Nothing} = VERSION,
         )
         # make sure all versions of all packages know about julia uuid
         for (uuid,vnmap) in deps, vn in versions[uuid]
             get!(valtype(vnmap), vnmap, vn)["julia"] = uuid_julia
         end
 
         # Tell the resolver about julia itself
-        fixed[uuid_julia] = Fixed(VERSION)
         uuid_to_name[uuid_julia] = "julia"
-        versions[uuid_julia] = Set([VERSION])
-        deps[uuid_julia] = DepsValDict(VERSION => valtype(DepsValDict)())
-        compat[uuid_julia] = CompatValDict(VERSION => valtype(CompatValDict)())
+        if julia_version !== nothing
+            fixed[uuid_julia] = Fixed(julia_version)
+            versions[uuid_julia] = Set([julia_version])
+            deps[uuid_julia] = DepsValDict(julia_version => valtype(DepsValDict)())
+            compat[uuid_julia] = CompatValDict(julia_version => valtype(CompatValDict)())
+        else
+            versions[uuid_julia] = Set([])
+        end
 
-        extra_uuids = union(keys(reqs), union(keys(fixed), map(fx->keys(fx.requires), values(fixed))...))
+        extra_uuids = union(collect(keys(reqs)), union(collect(keys(fixed)), map(fx->keys(fx.requires), values(fixed))...))
```

Comment:
Out of curiosity, what made this needed?

Reply:
If `fixed` is an empty array, then some codepath of `union()` ends up calling `copy(keys(fixed))`, and it complains that there is no `copy(::KeySet{...})` method.  You can recreate the fun yourself with the following snippet:


```
julia> using Pkg
       fixed = Dict{Base.UUID,Pkg.Resolve.Fixed}()
       union(keys(fixed))
ERROR: MethodError: no method matching copy(::Base.KeySet{Base.UUID,Dict{Base.UUID,Pkg.Resolve.Fixed}})
Closest candidates are:
  copy(::Pkg.Resolve.FieldValue) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/Pkg/src/Resolve/fieldvalues.jl:63
  copy(::Random._GLOBAL_RNG) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/Random/src/RNGs.jl:325
  copy(::Expr) at expr.jl:35
  ...
Stacktrace:
 [1] union(::Base.KeySet{Base.UUID,Dict{Base.UUID,Pkg.Resolve.Fixed}}) at ./abstractset.jl:49
 [2] top-level scope at REPL[12]:3
```

This is happening because the splat has zero elements.  This never happened before, because the `julia` entry ensured that at least one fixed element always existed.